tensorbay/opendataset/PlantSeedlings/loader.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+# pylint: disable=missing-module-docstring
+
+import os
+from typing import Dict, Iterator, Optional
+
+import numpy as np
+from PIL import Image
+
+from ...dataset import Data, Dataset, Segment
+from ...label import Classification, SemanticMask
+from .._utility import glob
+
+DATASET_NAME = "PlantSeedlings"
+
+_FULLIMAGES_MAP = {
+    (1, 4): "Maize",
+    (5, 8): "Common wheat",
+    (9, 12): "Sugar beet",
+    (13, 16): "scentless mayweed",
+    (17, 20): "Common Chickweed",
+    (21, 24): "shepherd's-purse",
+    (25, 28): "Cleavers",
+    (29, 32): "Fersken pileurt",
+    (33, 36): "Charlock",
+    (37, 40): "Fat Hen",
+    (41, 44): "Cranes-bill",
+    (45, 48): "Field Pansy",
+    (49, 52): "Black grass",
+    (53, 56): "Loose Silky-bent",
+}
+
+
+def PlantSeedlings(path: str) -> Dataset:
+    """Dataloader of PlantSeedlings dataset.
+
+    .. PlantSeedlings: https://vision.eng.au.dk/plant-seedlings-dataset/
+
+    The file structure should be like::
+
+        <path>
+            Fullimages/
+                1/
+                    IMG_0573_12-03.png
+                    ...
+                2/
+                    ...
+                ...
+                SpeciesIndex.pdf
+                SpeciesIndex.xlsx
+            Black-grass/
+                1.png
+                2.png
+                ...
+            Charlock/
+            Cleavers/
+            Common Chickweed/
+            Common wheat/
+            Fat Hen/
+            Loose Silky-bent/
+            Maize/
+            Scentless Mayweed/
+            Shepherd’s Purse/
+            Small-flowered Cranesbill/
+            Sugar beet/
+            Segmented/
+                Black-grass/
+                 1.png
+                 2.png
+                 ...
+                Charlock/
+                Cleavers/
+                Common Chickweed/
+                Common wheat/
+                Fat Hen/
+                Loose Silky-bent/
+                Maize/
+                Scentless Mayweed/
+                Shepherd’s Purse/
+                Small-flowered Cranesbill/
+                Sugar beet/
+
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    for subset in os.listdir(root_path):
+        if subset == "Segmented":
+            continue
+        segment = (
+            _get_segment(root_path, subset)
+            if subset != "Fullimages"
+            else _get_fullimages(root_path)
+        )
+        dataset.add_segment(segment)
+    return dataset
+
+
+def _get_fullimages(root_path: str) -> Segment:
```

Comment:
Should we put the Full image and Cropped images in one dataset?
Should we only keep the Cropped image or split Full images and Cropped images into two datasets?